src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpTraceSerializer.cs

```diff
@@ -15,6 +16,158 @@ internal static class ProtobufOtlpTraceSerializer
     private const int TraceIdSize = 16;
     private const int SpanIdSize = 8;
 
+    private static readonly ConcurrentBag<List<Activity>> ActivityListPool = [];
```

Comment:
I don't think a concurrent thing is needed here. Often the concurrent versions are a bit slower for reads vs non-concurrent versions because they need to do more work to be safe.

Reply:
Fixed.